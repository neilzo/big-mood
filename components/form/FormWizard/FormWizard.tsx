import React from 'react';
import { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';

export const isLastStep = (step: number, steps: Array<Element>) =>
  step === steps.length - 1;

interface Props {
  step: number;
  steps: Array<Element>;
  saveButton: Element;
  deleteButton?: Element | null;
}
interface State {
  step: number;
}
export default class FormWizard extends Component<Props, State> {
  static defaultProps = {
    step: 0,
  };

  constructor(props: Props) {
    super(props);

    this.state = {
      step: props.step,
    };
  }

  handleNextPress = () =>
    this.setState((prevState: State) => ({ step: prevState.step + 1 }));

  handleBackPress = () =>
    this.setState((prevState: State) => ({ step: prevState.step - 1 }));

  renderStep = () => {
    const { steps } = this.props;
    const { step } = this.state;

    return steps[step];
  };

  renderBackButton = () => {
    const { step } = this.state;

    if (step === 0) return null;

    return <Button title="Back" onPress={this.handleBackPress} />;
  };

  renderNextButton = () => {
    const { steps } = this.props;
    const { step } = this.state;
    const isEnd: boolean = isLastStep(step, steps);

    if (isEnd) return null;

    return <Button title="Next" onPress={this.handleNextPress} />;
  };

  renderSaveButton = () => {
    const { saveButton, steps } = this.props;
    const { step } = this.state;
    const isEnd: boolean = isLastStep(step, steps);

    if (!isEnd) return null;

    return saveButton;
  };

  renderDeleteButton = () => {
    const { deleteButton } = this.props;

    return deleteButton;
  };

  render() {
    return (
      <View style={styles.container}>
        {this.renderBackButton()}
        {this.renderNextButton()}
        {this.renderStep()}
        {this.renderBackButton()}
        {this.renderNextButton()}
        {this.renderSaveButton()}
        {this.renderDeleteButton()}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'stretch',
    alignItems: 'center',
  },
});
