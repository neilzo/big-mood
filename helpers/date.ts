import moment from 'moment';

const getPrettyDate = (date: Date) => moment(date).format('MMM Do YYYY');

const getPrettyTime = (date: Date) => moment(date).format('h:mma');

export default { getPrettyDate, getPrettyTime };
