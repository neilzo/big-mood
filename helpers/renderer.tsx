import 'react-native';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

export const createComponent = (component: any) => {
  return renderer.create(component);
};

// export const createComponentWithRedux = (children: any, props = {}) => {
//   return renderer.create(<Provider {...props}>{children}</Provider>);
// };
