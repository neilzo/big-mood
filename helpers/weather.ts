interface WeatherResults {
  weather: Array<{ main: string; icon: string }>;
  main: {
    temp: number;
  };
}

const kelvinToFarenheit = (kelvin: number) => {
  return Math.round(kelvin * (9 / 5) - 459.76);
};

const computeOverallWeather = (weatherResults: WeatherResults) => ({
  icon: weatherResults.weather[0].icon,
  description: weatherResults.weather[0].main,
  temperature: kelvinToFarenheit(weatherResults.main.temp),
});

export { kelvinToFarenheit, computeOverallWeather };
