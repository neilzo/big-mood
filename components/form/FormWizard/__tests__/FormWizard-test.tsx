import React from 'react';
import FormWizard, { isLastStep } from '../FormWizard';
import { View, Text } from 'react-native';
import { createComponent } from '../../../../helpers/renderer'; // todo resolve imports to avid that ugly thang

jest.mock('../../../Icon/Icon', () => 'Icon');

const mockProps = {
  steps: [<View />, <View />],
  saveButton: (
    <View>
      <Text>Save, dude</Text>
    </View>
  ),
};

describe('FormWizard', () => {
  it('renders correctly', () => {
    expect(
      createComponent(<FormWizard {...mockProps} />).toJSON()
    ).toMatchSnapshot();
  });

  it('renders with a delete button', () => {
    expect(
      createComponent(
        <FormWizard
          {...mockProps}
          deleteButton={
            <View>
              <Text>Delete meee</Text>
            </View>
          }
        />
      ).toJSON()
    ).toMatchSnapshot();
  });

  describe('isLastStep', () => {
    it('should return true if the step is the last', () => {
      expect(isLastStep(1, [{}, {}])).toBe(true);
    });

    it('should return false if the step is NOT the last', () => {
      expect(isLastStep(0, [{}, {}])).toBe(false);
    });
  });
});
