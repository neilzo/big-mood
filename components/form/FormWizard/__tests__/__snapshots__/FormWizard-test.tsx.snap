// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormWizard renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
  <View />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;

exports[`FormWizard renders with a delete button 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
  <View />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
  <View>
    <Text>
      Delete meee
    </Text>
  </View>
</View>
`;
